modules/apps/analytics/analytics-settings-web/src/main/java/com/liferay/analytics/settings/web/internal/portlet/action/BaseAnalyticsMVCRenderCommand.java

```diff
@@ -89,6 +89,9 @@ protected String getJspPath() {
 	@Reference
 	protected ConfigurationProvider configurationProvider;
 
+	@Reference(
+		target = "(osgi.web.symbolicname=com.liferay.analytics.settings.web)"
+	)
 	protected volatile ServletContext servletContext;
```

Comment:
@JanisZhang Please remove `volatile`.

According to the JavaDoc of `org.osgi.service.component.annotations.Reference#policy`, if the annotation is applied to a method, the default policy is `ReferencePolicy.STATIC`; if it's applied on a field, the default policy is `ReferencePolicy.DYNAMIC` if the field is `volatile`, and `ReferencePolicy.STATIC` otherwise.

Here the reference is moved from method (default STATIC) to a volatile field (default DYNAMIC), we should keep it STATIC.

cc @tinatian